libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/task/OfflineRouteRetrievalTask.kt

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.navigation.route.onboard.task
+
+import android.os.AsyncTask
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.mapbox.navigation.base.route.model.Route
+import com.mapbox.navigation.logger.MapboxLogger
+import com.mapbox.navigation.navigator.MapboxNativeNavigator
+import com.mapbox.navigation.route.onboard.OnOfflineRouteFoundCallback
+import com.mapbox.navigation.route.onboard.model.OfflineError
+import com.mapbox.navigation.route.onboard.model.OfflineRouteError
+import com.mapbox.navigator.RouterResult
+
+internal class OfflineRouteRetrievalTask(
+    private val navigator: MapboxNativeNavigator,
+    private val callback: OnOfflineRouteFoundCallback
+) : AsyncTask<String, Void, List<Route>>() {
```

Comment:
We should take advantage of this refactor and replace `AsyncTask` by something else (Coroutines?) as `AsyncTask`s are officially deprecated https://android-review.googlesource.com/c/platform/frameworks/base/+/1156409/6/core/java/android/os/AsyncTask.java

Reply:
I see your point, we could make it later because it's not hight priority